File: static/CWE-89/csharp/patched.cs
// Remediation for CWE-89: SQL Injection
// Fix: Use a parameterized SqlCommand with Parameters.AddWithValue instead of string concatenation.

using System.Data.SqlClient;

namespace GuineaPig.Sast
{
    class Patched_89
    {
        public void LookupUser(string userInput)
        {
            using (var conn = new SqlConnection("Server=db;Database=app;Integrated Security=true;"))
            {
                conn.Open();
                string query = "SELECT * FROM Users WHERE Name = @name";
                using (var cmd = new SqlCommand(query, conn))
                {
                    cmd.Parameters.AddWithValue("@name", userInput);
                    using (var reader = cmd.ExecuteReader())
                    {
                        while (reader.Read())
                        {
                            System.Console.WriteLine(reader[0]);
                        }
                    }
                }
            }
        }
    }
}

File: static/CWE-89/java/patched.java
// Remediation for CWE-89: SQL Injection
// Fix: Use a PreparedStatement with a bound parameter instead of string concatenation.

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;

class Patched_89 {
    public ResultSet findUser(String userInput) throws Exception {
        Connection conn =
            DriverManager.getConnection("jdbc:mysql://localhost/app", "app", "app");
        PreparedStatement stmt =
            conn.prepareStatement("SELECT * FROM users WHERE name = ?");
        stmt.setString(1, userInput);
        return stmt.executeQuery();
    }
}
